packages/mui-base/src/Dialog/Trigger/useDialogTrigger.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { useDialogRootContext } from '../Root/DialogRootContext';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import type { UseDialogTriggerReturnValue } from './DialogTrigger.types';
+/**
+ *
+ * Demos:
+ *
+ * - [Dialog](https://mui.com/base-ui/react-dialog/#hooks)
+ *
+ * API:
+ *
+ * - [useDialogTrigger API](https://mui.com/base-ui/react-dialog/hooks-api/#use-dialog-trigger)
+ */
+export function useDialogTrigger(): UseDialogTriggerReturnValue {
+  const { open, onOpenChange, modal, popupElementId } = useDialogRootContext();
+
```

Comment:
I don't think hooks should use context implicitly, rather explicit params passed into them as an object argument.  Context providers should be the responsibility of the component author using the hooks. So context types/providers aren't exported publicly.